docs/source/developers/guide/step_by_step/pr_and_github.rst

```diff
@@ -24,14 +24,171 @@
 
 .. _pr_and_github:
 
-****************************
-Lifecycle of a Pull Request
-****************************
+******************************
+Lifecycle of a Pull Request 🙀 
+******************************
 
+:ref:`As mentioned before<set-up>`, the Arrow project uses Git for
+version control and a workflow based on Pull Requests. That means
+that you contribute the changes to the code by creating a branch
+in Git, make changes to the code, push the changes to your ``origin``
+which is your fork of the Arrow repository on GitHub and then you
+create a **Pull Request** against the official Arrow repository
+which is saved in your set up as ``upstream``.
 
-Creating a PR 🙀
-================
+You should have Git set up by now, have cloned the repository,
+have successfully built Arrow and have an JIRA issue to work on.
 
+**Before making changes to the code, you should create a new
+branch in Git.**
 
-Reviews and get the PR merge 🎉
-===============================
\ No newline at end of file
+1. Update/sync the code from your ``upstream``
+   in the master branch. Run it in the shell from ``arrow`` directory.
+
+   .. code:: console
+
+      $ git checkout master # select the main Arrow branch
+      $ git fetch upstream # check for changes in upstream/master
+      $ git pull --ff-only upstream master # save the changes from upstream/master
+
+   Note: ``--ff-only`` applies changes only if they can be fast-forwarded
+   without creating new commits.
+
+2. Create a new branch
+
+   .. code:: console
+
+      $ git checkout -b <branch-name>
+
+   or (does the same thing)
+   
+   .. code:: console
+
+      $ git switch --create <branch-name>
+
+Now you can make changes to the code. To see the changes
+made in the library use this two commands:
+
+.. code:: console
+
+   $ git status # to see what files are changed
+   $ git diff # to see code change per file
+
+Creating a Pull Request 
+=======================
+
+Once you are satisfied with the changes, run the :ref:`tests <testing>`
+and linters and then go ahead and commit the changes.
+
+3. Add and commit the changes
+
+   .. code:: console
+         
+      $ git add <filenames>
+      $ git commit -m '<message>'
```

Comment:
```suggestion
      $ git commit -m "<message>"
```

(I generally find double quotes a bit safer in case you would use ' as apostrophe for some reason in the message)